app/js/month-view.jsx

```diff
@@ -20,30 +20,67 @@ class MonthView extends React.Component {
 
     renderWeeks() {
         // start at the first of the month
-        var currentDate = moment(this.props.date).date(1).hour(0).minute(0).second(0).milliseconds(0),
+        const now = moment.tz(this.props.timezone);
+        var currentDate = this.props.date.clone().date(1).hour(0).minute(0).second(0).milliseconds(0),
             currentMonth = currentDate.month();
 
-        var weeks = [], i = 0;
-
-        while(currentDate.month() === currentMonth) {
-            var dayOfWeek = currentDate.day(), dateToPass = moment(currentDate);
+        const weeks = [];
+        let i = 0;
 
 
+        while(currentDate.month() <= currentMonth) {
+            let dayOfWeek = currentDate.day();
 
+            // if it's the first day of the month
             if(i === 0) {
+                // and it's not sunday, we want to go back to the previous sunday to start
                 if(dayOfWeek !== 0) {
-                    dateToPass = moment(currentDate).subtract(currentDate.day(), "days");
+                  // go back to sunday
+                  currentDate.subtract(currentDate.day(), "days");
                 }
             }
-
-
+            // if it's a sunday, or it's the first, render a week
             if(dayOfWeek === 0 || i === 0) {
-                 weeks.push(<WeekRow selectedDate={this.props.selectedDate} maxDate={this.props.maxDate} minDate={this.props.minDate} handleSelection={this.props.handleSelection} month={currentMonth} date={dateToPass} key={currentDate.date()} />);
+              let dates = [];
+
+              dates.push({
+                inMonth: currentDate.month() === currentMonth,
+                today: currentDate.format("MM/DD/YYYY") === now.format("MM/DD/YYYY"),
+                selected: this.props.selectedDate && currentDate.format("MM/DD/YYYY") === this.props.selectedDate.format("MM/DD/YYYY"),
+                disabled: currentDate.isBefore(this.props.minDate) || currentDate.isAfter(this.props.maxDate),
+                date: currentDate.date(),
+                formattedDate: currentDate.format("MM/DD/YYYY")
+              });
+
+              while(currentDate.day() < 6) {
```

Comment:
This all seems to work as far as I can tell, but I don't understand the logic behind all of these changes 🙃 . Could you just document the reasons why we sometimes need to add/subtract a day here?